pandas/core/generic.py

```diff
@@ -3151,150 +3167,338 @@ def to_latex(
+        \toprule
+        name & mask & weapon \\
+        \midrule
+        Raphael & red & sai \\
+        Donatello & purple & bo staff \\
         \bottomrule
         \end{{tabular}}
         """
-        # Get defaults from the pandas config
-        if self.ndim == 1:
-            self = self.to_frame()
-        if longtable is None:
-            longtable = config.get_option("display.latex.longtable")
-        if escape is None:
-            escape = config.get_option("display.latex.escape")
-        if multicolumn is None:
-            multicolumn = config.get_option("display.latex.multicolumn")
-        if multicolumn_format is None:
-            multicolumn_format = config.get_option("display.latex.multicolumn_format")
-        if multirow is None:
-            multirow = config.get_option("display.latex.multirow")
-
-        self = cast("DataFrame", self)
-        formatter = DataFrameFormatter(
-            self,
-            columns=columns,
-            col_space=col_space,
+        kwargs = locals()
+        deprecated_args = {
+            "longtable": '`environment="longtable"`',
+            "multicolumn": "`multicol_align`",
+            "multicolumn_format": "`multicol_align`",
+            "multirow": "`multirow_align`",
+            "col_space": None,
+            "formatters": "`formatter`, `precision`, `decimal`, `thousands`",
+            "float_format": "`precision`, `decimal`, `thousands`, `formatter`",
+            "sparsify": "`sparse_index`, `sparse_columns`",
+            "bold_rows": "`bold_header`",
+        }
+        for k, v in deprecated_args.items():
+            if kwargs[k] is not None:
+                warnings.warn(
+                    f"`{k}` is deprecated after transition to `Styler.to_latex()` "
+                    f"signature." + ("" if v is None else f" Consider {v} instead."),
+                    FutureWarning,
+                    stacklevel=2,
+                )
+
+        from pandas.io.formats.style import Styler
+
+        if (
+            escape is None and get_option("styler.format.escape") == "latex"
+        ) or escape is True:
+            escape = "latex"
+        else:
+            escape = None
+        # error: Argument 1 to "Styler" has incompatible type "NDFrame"; expected
+        # "Union[DataFrame, Series]"
+        styler = Styler(
+            self,  # type: ignore[arg-type]
+            uuid="",
+            formatter=formatter,
             na_rep=na_rep,
-            header=header,
-            index=index,
-            formatters=formatters,
-            float_format=float_format,
-            bold_rows=bold_rows,
-            sparsify=sparsify,
-            index_names=index_names,
-            escape=escape,
+            precision=precision,
             decimal=decimal,
+            thousands=thousands,
+            escape=escape,
         )
-        return DataFrameRenderer(formatter).to_latex(
+
+        for ax in [0, 1]:  #
```

Comment:
Everything below this line contains quite complex logic.
I suppose that extracting a whole new class, like ``StylerLatexAdapter`` or something like this, may be reasonable here.
It will then be possible to encapsulate kwargs deprecation, handling headers, index, etc there.

Reply:
There is a gap in `Styler` currently since `format_index` is so new, that a `Styler` class can be instantiated with formatting variable for data objects, but not for index objects. Of course it opens up a query how best to do this since there are quite a lot of formatting args and if applied to data values and each index separately it introduces a potential host of new args.

With regards to the general complexity, yes I agree the logic has been to squeeze one method into the other and an adapter might be useful. Can we wait on feedback from others too?